sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/healthcare/FhirIO.java

```diff
@@ -155,17 +168,53 @@
  * FhirIO.Write.Result writeResult =
  *     output.apply("Execute FHIR Bundles", FhirIO.executeBundles(options.getExistingFhirStore()));
  *
+ * // Alternatively you could use import for high throughput to a new store.
+ * FhirIO.Write.Result writeResult =
+ *     output.apply("Import FHIR Resources", FhirIO.executeBundles(options.getNewFhirStore()));
+ * // [End Writing ]
+ *
  * PCollection<HealthcareIOError<String>> failedBundles = writeResult.getFailedInsertsWithErr();
  *
+ * // [Begin Writing to Dead Letter Queue]
  * failedBundles.apply("Write failed bundles to BigQuery",
  *     BigQueryIO
  *         .write()
  *         .to(option.getBQFhirExecuteBundlesDeadLetterTable())
  *         .withFormatFunction(new HealthcareIOErrorToTableRow()));
+ * // [End Writing to Dead Letter Queue]
+ *
+ * // Alternatively you may want to handle DeadLetter with conditional update
+ * // [Begin Reconciliation with Conditional Update]
+ * failedBundles
+ *     .apply("Reconcile with Conditional Update",
+ *         FhirIO.ConditionalUpdate(fhirStore)
+ *             .withFormatBodyFunction(HealthcareIOError<String>::getDataResource)
+ *             .withTypeFunction((HealthcareIOError<String> err) -> {
+ *               String body = err.getDataResource();
+ *               // TODO(user) insert logic to exctract type.
+ *               return params;
+ *             })
+ *             .withSearchParametersFunction((HealthcareIOError<String> err) -> {
```

Comment:
It seems like the search parameters follow [a particular syntax](https://cloud.google.com/healthcare/docs/reference/rest/v1beta1/projects.locations.datasets.fhirStores.fhir/search#google.cloud.healthcare.v1beta1.fhir.rest.FhirService.SearchResources) - does it make sense to provide a builder class for search parameters? this may help prevent mistakes on building the query

e.g:
```
FhirSearchParameter.builder()
  .withBase(...)
  .withType(FhirIO.ResourceType.Patient)
  .addParameter(...)
  .withSearchModifier(...)
  .toString(); // (or .build())
```

The con is that it may add the burden of having to add support for every new thing added to the search parameter query. It seems a bit complex, so IDK. Thoughts?

Reply:
this is a great suggestion will add a FhirSearchParameter Builder.

Reply:
Hmm on second thought the best way to do this if for the user to use HAPI library search builder which will be better maintained. 
I think we can add this as a suggestion in the docs rather than making this a beam dependency or duplicating the functionality. Thoughts?
https://hapifhir.io/hapi-fhir/docs/server_plain/rest_operations_search.html

Reply:
I think that makes total sense. Thanks!